src/app/screens/TimesheetScreen/view/createTimesheet.tsx

```diff
@@ -63,73 +63,76 @@ const CreateTimesheet = ({toggleModal, isVisible}: Props) => {
     };
   }, []);
 
-  const onAdd = (data?: TimesheetFormData, resetField?: Function) => {
-    const reset = () => {
-      setFormDefaultData(undefined);
-
-      if (typeof resetField !== 'undefined') {
-        resetField('project');
-        resetField('date');
-        resetField('workHours');
-        resetField('description');
-      }
-      handlePress(false);
-
-      Keyboard.dismiss();
-    };
-
-    if (data) {
-      let isCategoryFound = false;
-      setAddedTimesheet(sections => {
-        sections.forEach(section => {
-          if (section.title === data.project) {
-            let isPresent = false;
-            isCategoryFound = true;
-
-            section.data.forEach(item => {
-              if (
-                item.timesheet_id ===
-                data.project + dateFormater(data.date)
-              ) {
-                isPresent = true;
+  const onAdd = useCallback(
+    (data?: TimesheetFormData, resetField?: Function) => {
+      const reset = () => {
+        setFormDefaultData(undefined);
+
+        if (typeof resetField !== 'undefined') {
+          resetField('project');
+          resetField('date');
+          resetField('workHours');
+          resetField('description');
+        }
+        handlePress(false);
+
+        Keyboard.dismiss();
+      };
+
+      if (data) {
+        let isCategoryFound = false;
+        setAddedTimesheet(sections => {
+          sections.forEach(section => {
+            if (section.title === data.project) {
+              let isPresent = false;
+              isCategoryFound = true;
+
+              section.data.forEach(item => {
+                if (
+                  item.timesheet_id ===
+                  data.project + dateFormater(data.date)
+                ) {
+                  isPresent = true;
+                }
+              });
```

Comment:
Instead of this we can check the condition using find
```suggestion
              const isTimesheetPresent = section.data.find(item => {
                return (
                  item.timesheet_id ===
                  data.project + dateFormater(data.date)
                )
              });
```